src/subdomains/payout/strategies/payout/impl/base/jellyfish.strategy.ts

```diff
@@ -94,10 +94,9 @@ export abstract class JellyfishStrategy extends PayoutStrategy {
     const result: Map<number, PayoutOrder[]> = new Map();
 
     orders.forEach((o) => {
-      // find nearest non-full group without repeating address
+      // find nearest non-full group
       const suitableExistingGroups = [...result.entries()].filter(
-        ([_, _orders]) =>
-          _orders.length < maxGroupSize && !_orders.find((_o) => _o.destinationAddress === o.destinationAddress),
+        ([_, _orders]) => new Set(_orders.map((_o) => _o.destinationAddress)).size < maxGroupSize,
       );
```

Comment:
This will create a new group, if all existing groups are full, even if there is a group which contains same destination address.